### ISSUE TITLE
Stuck on "[DEBUG] Slow down and retry" on ID not found

### ISSUE THREAD

**Issue by alan-hasegawa** [RicterZ/nhentai#188]
Version 0.4.9, the program gets stuck fetching information of a doujinshi that does not exist, as shown below when trying to fetch ID 200000. The same behavior was observed when setting the cookies.

```
D:\nhentai\nhentai-master>nhentai --id="200000"
[16:56:37] [INFO] nHentai ver 0.4.9: あなたも変態。 いいね?
       _   _            _        _
 _ __ | | | | ___ _ __ | |_ __ _(_)
| '_ \| |_| |/ _ \ '_ \| __/ _` | |
| | | |  _  |  __/ | | | || (_| | |
|_| |_|_| |_|\___|_| |_|\__\__,_|_|

[16:56:37] [INFO] Using mirror: https://nhentai.net
[16:56:37] [INFO] Using viewer template "default"
[16:56:38] [ERROR] Cannot get your username, please check your cookie or use `nhentai --cookie` to set your cookie
[16:56:38] [INFO] Fetching doujinshi information of id 200000
[16:56:38] [DEBUG] Slow down and retry (200000) ...
[16:56:39] [INFO] Fetching doujinshi information of id 200000
[16:56:40] [DEBUG] Slow down and retry (200000) ...
[16:56:41] [INFO] Fetching doujinshi information of id 200000
[16:56:42] [DEBUG] Slow down and retry (200000) ...
[16:56:42] [INFO] Fetching doujinshi information of id 200000
[16:56:43] [ERROR] Ctrl-C signal received. Stopping...

```

Apparently, there are some searches that return doujinshis that don't exist, for example: 
```
D:\nhentai\nhentai-master>nhentai --search="lolishota"
[17:08:52] [INFO] nHentai ver 0.4.9: あなたも変態。 いいね?
       _   _            _        _
 _ __ | | | | ___ _ __ | |_ __ _(_)
| '_ \| |_| |/ _ \ '_ \| __/ _` | |
| | | |  _  |  __/ | | | || (_| | |
|_| |_|_| |_|\___|_| |_|\__\__,_|_|

[17:08:52] [INFO] Using mirror: https://nhentai.net
[17:08:52] [INFO] Using viewer template "default"
[17:08:52] [ERROR] Cannot get your username, please check your cookie or use `nhentai --cookie` to set your cookie
[17:08:52] [INFO] Searching doujinshis using keywords "lolishota" on page 1
[17:08:55] [INFO] Search Result || Found 25 doujinshis
======  =======================================================================================
    id  doujinshi
======  =======================================================================================
342901  [Shinsei Lolishota (kozi)] Hentai Inmon Otokonoko Massage [Chinese] [管少女汉化] [Digital]
342047  [Sinsei Loli Shota (Hokuro Ryuseigun)] Netorare Kareshi ~Hamerareshi Bishojo~
341029  [新生ロリショタ (kozi、幸無ろき雄)] メスガキ盛り男の娘 誘惑下着で下心
340106  [Shinsei Lolishota (kozi)] Kyuusei! Otokonoko Succubus-kun [Digital]
339582  [Shinsei Lolishota (kozi)] Akogare no Elf Shounen ga Yowatte ita node Chiryou Tsuide ..
(etc...)
```
I get the following http status codes for the IDs in the search result:
```
342901 returns 200 OK
342047 returns 200 OK
341029 returns 404 Not Found
340106 returns 404 Not Found
339582 returns 404 Not Found
```

So it seems that nhentai's search results may point to an invalid id and the program gets stuck in a loop trying to fetch it.

I hope the description is detailed enough to reproduce the issue. 

This is some great work btw. Thank you for making this.

-Alan

**Comment by mobrine1** [comment]
I'm not able to reproduce issue of your second command.
It seems like 341029 340106 339582 have been deleted

**Comment by alan-hasegawa** [comment]
> I'm not able to reproduce issue of your second command.
> It seems like 341029 340106 339582 have been deleted from website itself between the time you used the command and looked if id exists

Thank you for the reply .

Same, now it's returning the two valid doujinshi. Maybe the search results were from an old cache after the other ones got deleted.

First issue can still be reproduced, trying to fetch an id that has no doujinshi makes the program stuck in a loop

-Alan

**Comment by RicterZ** [comment]
fixed in 0.4.10
